File: 98-2 查找和替换模式/52ms.py
class Solution(object):
    def findAndReplacePattern(self, words, pattern):
        """
        :type words: List[str]
        :type pattern: str
        :rtype: List[str]
        """
        d={}
        dd={}
        l=[]
        for i in words:
            d.clear()
            dd.clear()
            flag=True
            for j in range(len(pattern)):
                if pattern[j] in d and d[pattern[j]]!=i[j]:
                    flag=False
                    break
                if i[j] in dd and dd[i[j]]!=pattern[j]:
                    flag=False
                    break
                d[pattern[j]]=i[j]
                dd[i[j]]=pattern[j]
            if flag:l.append(i)
        return l

File: 137-2 删除字符串中的所有相邻重复项/52ms.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        int l=S.size();
        bool flag=true;
        while(flag){
            int len=l;
            string s="";
            flag=false;
            for(int i=0;i<l;){
                if(i<l-1&&S[i]==S[i+1]){
                    flag=true;
                    i+=2;
                    len-=2;
                }else{
                    s+=S[i++];
                }
            }
            S=s;
            l=len;
        }
        return S;
    }
};